File: arrays/two_sum/two_sum.py
from networkx import complement


def two_sum(nums: list[int], target: int) -> tuple[int, int]:
    """
    Problem: Two Sum
    Difficulty: Easy
    Category: Arrays
    
    Find two numbers that add up to target and return their indices.
    """
    print(f"Input: nums = {nums}, target = {target}")
    
    num_map = {}
    for i, num in enumerate(nums):
        output = target - num
        if output in num_map:
            print(f"Output: [{num_map[output]}, {i}]")
            return (num_map[output], i)
        num_map[num] = i
    
    print("No two sum solution found.")
    return (-1, -1)

if __name__ == "__main__":
    test_cases = [
        {"nums": [2, 7, 11, 15], "target": 9},
        {"nums": [3, 2, 4], "target": 6},
        {"nums": [3, 3], "target": 6},
    ]
    
    for tc in test_cases:
        two_sum(tc["nums"], tc["target"])
    
    print("Python solution ran successfully!")

File: arrays/two_sum/two_sum.go
package main

import "fmt"

// solve runs the solution logic
func twoSum(nums []int, target int) (int, int) {

	m := make(map[int]int)
	for i, num := range nums {
		sub := target - num
		if j, ok := m[sub]; ok {
			return j, i
		}
		m[num] = i
	}
	return -1, -1
}

func main() {
	testCases := []struct {
		nums   []int
		target int
	}{
		{[]int{2, 7, 11, 15}, 9},
		{[]int{3, 2, 4}, 6},
		{[]int{3, 3}, 6},
	}

	for _, tc := range testCases {
		i, j := twoSum(tc.nums, tc.target)
		fmt.Printf("twoSum(%v, %d) = (%d, %d)\n", tc.nums, tc.target, i, j)
	}
}
